Pruning of irrelevant files during the finish step

adiosnb opened the pull request:
The patch deletes a plan worktree and also removes all non-YAML files from the whole plan directory
to reduce the amount of stored data.

Closes #1707

psss commented on tmt/steps/finish/__init__.py:
```diff
@@ -135,6 +149,24 @@ def go(self) -> None:
+                f'find {self.plan.workdir} -mindepth 2 -maxdepth 2 ' +
+                ' '.join([f'| grep -v {x}' for x in self._preserved_files]) +
+                '| xargs rm -rvf',
+                shell=True)
```
I'm not sure if grepping for `yaml` and `data` is the best approach here. The directories and fetched git repositories can contain a bunch of such files (including the `tmt` repo itself).

I was thinking about a `clean()` method which each `Step` would implement and which would take care of the cleanup and keeping the interesting files around. Or, maybe, `BasePlugin.cleanup()` method which would by default do a `shutil.rmtree(self.workdir)`?

@adiosnb, what do you think?

adiosnb replied:
I will add a generic clean() method and extends it if necessary to preserve custom files.

psss replied:
Thanks, this looks much better now!

psss commented on tmt/steps/finish/__init__.py:
```diff
@@ -50,6 +51,15 @@ def finish(context: 'tmt.cli.Context', **kwargs: Any) -> None:
+        return [
+            click.option(
+                '-k', '--keep', is_flag=True,
+                help='Keep all test files rsynced to local host (Skip pruning).'
```
Originally I thought the proper place for the new `--keep` option would be the `run` command as it is similar to what `--remove` does there:

    Options:
    -i, --id ID                     Run id (name or directory path).
    -l, --last                      Execute the last run once again.
    -r, --rm, --remove              Remove the workdir when test run is finished.

But here it also make sense. @happz, @lukaszachy, any recommendations here?

lukaszachy replied:
Personally I'd prefer to have it as `run` option, otherwise I would need to add finish step on the commandline most of the times.

adiosnb replied:
I've been modifying some tests analyzing created files, and adding this parameter to them. And yes, it is annoying to write `finish` everywhere. Thus, I moved it under `run` as you suggested.  It is more convenient.

psss replied:
Yes, good point, under `run` it will be more handy.

psss reviewed (approved):
Thanks for addressing all comments. Now looks good. Added a few adjustments in 0e0d5cba. Please, have a look if everything's ok.

adiosnb commented:
> Thanks for addressing all comments. Now looks good. Added a few adjustments in [0e0d5cb](https://github.com/teemtee/tmt/commit/0e0d5cba944601b70ca3b2cf8672c73854380402). Please, have a look if everything's ok.

It seems to be OK to me. Let's wait for the test results.
